schema/google/showcase/v1beta1/compliance.proto

```diff
@@ -0,0 +1,147 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/api/client.proto";
+
+package google.showcase.v1beta1;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1beta1";
+option java_multiple_files = true;
+
+// This service is used to test that GAPICs handle various data types and HTTP annotations
+// correctly, which is particularly important for server interactions using REST rather than
+// HTTP. This service will also test that legacy, non-AIP-compliant patterns are consistently
+// supported by all GAPIC generators.
+service Compliance {
+  // This service is meant to only run locally on the port 7469 (keypad digits
+  // for "show").
+  option (google.api.default_host) = "localhost:7469";
+
+  // This method echoes the datapack request. This method exercises all parameters in the body
+  rpc RepeatDataBody(RepeatRequest) returns (RepeatResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/repeat:body"
+      body: "*"
+    };
+  }
+  // This method echoes the datapack request. This method exercises all parameters in query params
+  rpc RepeatDataQuery(RepeatRequest) returns (RepeatResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/repeat:query"
+    };
+  }
+}
+
+message RepeatRequest {
+  string name = 1;
+  ComplianceData info = 2;
+}
+
+message RepeatResponse {
+  ComplianceData info = 1;
+}
+
+
+// ComplianceData is a message used for testing REST transcoding of
+// different data types.
+message ComplianceData {
+
+  // scalar types
+
+  string f_string = 2;
+
+  int32 f_int32 = 3;
+  sint32 f_sint32 = 4;
+  sfixed32 f_sfixed32 = 5;
+
+  uint32 f_uint32 = 6;
+  fixed32 f_fixed32 = 7;
+
+  int64 f_int64 = 8;
+  sint64 f_sint64 = 9;
+  sfixed64 f_sfixed64 = 10;
+
+  uint64 f_uint64 = 11;
+  fixed64 f_fixed64 = 12;
+
+  double f_double = 13;
+  float f_float = 14;
+
+  bool f_bool = 15;
+
+  bytes f_bytes = 16;
+
+  // optional fields
+
+  optional string p_string = 17;
+  optional int32 p_int32 = 18;
+  optional double p_double = 19;
+  optional bool p_bool = 20;
+}
+
+message ComplianceGroup {
+  string name = 1;
+  repeated string rpcs = 2;
+  repeated RepeatRequest cases = 3;
+
+}
+
+message ComplianceSuite {
+  repeated ComplianceGroup group = 1;
+}
+
+
+
+// DataPack is a message used for testing REST transcoding of
+// different data types. In the future, it may be part of an Echo
+// service RPC that also tests JSON responses.
+message DataPack {
+  DataPack subpack = 1;  // recursive message cause problems (infinite loop?) for go_cli
```

Comment:
I can take a look at that. If we need to unblock this PR, can we just add this later?

Reply:
I defined `DataPack` in an earlier PR as part of `echo.proto`, and I've been using it in other tests that are checked in. That's why I haven't removed it yet. 

I added `ComplianceData` in this PR to get around the recursion issue. If it can be fixed, then we certainly only need one of the two protos. 

Do you think the recursion issue is fixable? I haven't explored it.

I was using recursion as a simple way of testing that I can transcode nested fields. I can work around that by adding non-recursive sub-messages to to `ComplianceData`, and switch everything over to use that instead of `DataPack`

Reply:
Well, it could be fixed in the gen_cli, I'd just limit the depth of the recursion (probably to just one), but its a low priority thing and if we could have some simple separate message to test nested messages, then I think that would would be preferred. I think your suggestion of using non-recursive sub-message would be good, if you are OK with that.

Reply:
Yeah, I'll do that.

Reply:
Done